src/mon/Elector.cc

```diff
@@ -459,8 +459,20 @@ void Elector::begin_peer_ping(int peer)
     return;
   }
 
-  if (!mon->get_quorum_mon_features().contains_all(
-				      ceph::features::mon::FEATURE_PINGING)) {
+  if (mon->get_quorum_mon_features().empty()) {
+    // quorum_mon_features has not been assigned yet.
+    // Sometimes at bootup we haven't finish our first election yet.
+    // Lets check if our feature supports FEATURE_PINGING.
+    mon_feature_t mon_supported = ceph::features::mon::get_supported();
+    if (!mon_supported.contains_all(ceph::features::mon::FEATURE_PINGING)) {
+      dout(10) << "We currently do not support FEATURE_PINGING"
+        << " please upgrade! ... return" << dendl;
+      return;
+    }
+  } else if (!mon->get_quorum_mon_features().contains_all(
+                                      ceph::features::mon::FEATURE_PINGING)) {
+    dout(10) << "Not all monitors support FEATURE_PINGING,"
+      << " please upgrade! ... return" << dendl;
```

Comment:
Oooh, I don't think this works. Having this code means that our monitor *does* support pinging. The problem is making sure that all the other monitors do — we can't send unrecognized messages. (I believe our peers will crash if we do. Not ideal, but monitors are highly trusted.)

Reply:
I see, okay it's just that sometimes when we freshly install the cluster, peer informations of the highest rank monitor is missing because we have never formed a quorum at epoch 1, so get_quorum_mon_features() is null since it is not yet assigned and so pinging is rejected for the highest monitor. The missing peer information definitely gets added back eventually through new elections. If this missing peer information at the start is acceptable, then I'll just make this PR to only deal with rejecting -1 ranks

Reply:
Why is the highest monitor the only one that we're missing? IIRC we evaluate this again once elections happen and so everybody should start pinging everybody at that point. If we're missing one monitor somehow...I mean that sounds like a fencepost error, but the only real distinction on elections is <leader> and <not_leader>.